tests/queries/0_stateless/00463_long_sessions_in_http_interface.sh

```diff
@@ -1,113 +1,87 @@
+# An infinite loop is required to make the test reliable. We will check that the timeout corresponds to the observed time at least once
+while true
+do
+    (
+        ${CLICKHOUSE_CURL} -sS "${CLICKHOUSE_URL}&session_id=${CLICKHOUSE_DATABASE}_7&session_timeout=1" --data-binary "SELECT 1"
+        ${CLICKHOUSE_CURL} -sS "${CLICKHOUSE_URL}&session_id=${CLICKHOUSE_DATABASE}_7&session_check=1" --data-binary "SELECT 1"
+        sleep 3
+        ${CLICKHOUSE_CURL} -sS "${CLICKHOUSE_URL}&session_id=${CLICKHOUSE_DATABASE}_7&session_check=1" --data-binary "SELECT 1" | grep -c -F 'Session not found'
+    ) | tr -d '\n' | grep -F '111' && break || sleep 1
 done
-sleep 3
-check "$url$session&session_check=1" "$select" "Exception.*Session not found" 1 "Session did not expire on time."
 
-create_temporary_table "$url$session&session_timeout=2"
-for _ in $(seq 1 5); do
-    check "$url$session&session_timeout=2" "$select_from_non_existent_table" "Exception.*Table .* doesn't exist." 1 "Session expired too early."
-    sleep 1
+echo "A session successfully expire after a timeout and the session's temporary table shadows the permanent table:"
+# An infinite loop is required to make the test reliable. We will check that the timeout corresponds to the observed time at least once
+${CLICKHOUSE_CLIENT} --multiquery --query "DROP TABLE IF EXISTS t; CREATE TABLE t (s String) ENGINE = Memory; INSERT INTO t VALUES ('World');"
+while true
+do
+    (
+        ${CLICKHOUSE_CURL} -X POST -sS "${CLICKHOUSE_URL}&session_id=${CLICKHOUSE_DATABASE}_8&session_timeout=1" --data-binary "CREATE TEMPORARY TABLE t (s String)"
+        ${CLICKHOUSE_CURL} -X POST -sS "${CLICKHOUSE_URL}&session_id=${CLICKHOUSE_DATABASE}_8" --data-binary "INSERT INTO t VALUES ('Hello')"
+        ${CLICKHOUSE_CURL} -sS "${CLICKHOUSE_URL}&session_id=${CLICKHOUSE_DATABASE}_8" --data-binary "SELECT * FROM t"
+        sleep 3
+        ${CLICKHOUSE_CURL} -sS "${CLICKHOUSE_URL}&session_id=${CLICKHOUSE_DATABASE}_8" --data-binary "SELECT * FROM t"
+    ) | tr -d '\n' | grep -F 'HelloWorld' && break || sleep 1
 done
-check "$url$session&session_timeout=2" "$select_from_temporary_table" "Hello" 1 "Session expired too early. Failed to update timeout in case of exceptions."
-sleep 4
-check "$url$session&session_check=1" "$select" "Exception.*Session not found" 1 "Session did not expire on time."
+${CLICKHOUSE_CLIENT} --multiquery --query "DROP TABLE t"
 
+echo "A session cannot be used by concurrent connections:"
+
+${CLICKHOUSE_CURL} -sS -X POST "${CLICKHOUSE_URL}&session_id=${CLICKHOUSE_DATABASE}_9&query_id=${CLICKHOUSE_DATABASE}_9" --data-binary "SELECT count() FROM system.numbers" >/dev/null &
+
+# An infinite loop is required to make the test reliable. We will ensure that at least once the query on the line above has started before this check
+while true
+do
+    ${CLICKHOUSE_CLIENT} --query "SELECT count() > 0 FROM system.processes WHERE query_id = '${CLICKHOUSE_DATABASE}_9'" | grep -c -F '1' && break || sleep 1
```

Comment:
This is flaky. 
If I understand your logic right here should be just grep -F '1'. Without -c option. This is busy loop which produces output only when it stops.
I will make an adjustment in place.